server/src/module-index.ts

```diff
@@ -0,0 +1,169 @@
+import * as path from 'path';
+import * as fs from 'fs';
+
+const RSVP = require('rsvp');
+const i = require('i')();
+
+const readdir = RSVP.denodeify(fs.readdir);
+const lstat = RSVP.denodeify(fs.lstat);
```

Comment:
I actually prefer https://github.com/sindresorhus/pify over `rsvp` for this since it's a smaller package and uses native Promises.